lib/Schedule/LongSteps/Storage/DynamoDB.pm

```diff
@@ -0,0 +1,538 @@
+        AttributeDefinitions => [
+            { AttributeName => 'id', AttributeType => 'S' },
+            ## Note those are only there for reference.
+            ## as AttributeDefinition must only defined attributes
+            ## used in the KeySchema
+            # { AttributeName => 'process_class', AttributeType => 'S' },
+            # { AttributeName => 'mstatus', AttributeType => 'S' },
+            # { AttributeName => 'what', AttributeType => 'S' },
+            { AttributeName => 'run_at_day' , AttributeType => 'S' }, # ISO8601 YYYY-MM-DD
+            { AttributeName => 'run_at', AttributeType => 'S' }, # ISO8601 YYYY-MM-DDTHH:MM:SS.000
+            # { AttributeName => 'run_id', AttributeType => 'S' }, # The current run_id
+            # { AttributeName => 'mstate', AttributeType => 'S' },
+            # { AttributeName => 'merror', AttributeType => 'S' },
+        ],
+        KeySchema => [
+            { AttributeName => 'id', KeyType => 'HASH' },
+        ],
+        GlobalSecondaryIndexes => [
+            {
+                IndexName => 'by_run_at_day',
+                KeySchema => [
+                    { AttributeName => 'run_at_day', KeyType => 'HASH' },
+                    { AttributeName => 'run_at', KeyType => 'RANGE' },
+                ],
+                Projection => {
+                    NonKeyAttributes => [ 'run_id' ],
+                    ProjectionType => 'INCLUDE'
+                },
+                ProvisionedThroughput => {
+                    ReadCapacityUnits => 2,
+                    WriteCapacityUnits => 2,
+                }
+            }
+        ],
+        ProvisionedThroughput => {
+            ReadCapacityUnits => 2,
+            WriteCapacityUnits => 2,
+            # This is low to avoid having a large bill in case
+            # of tests failure.
+            # Note that we can change that AFTER the fact
+        },
+    );
+
+    while(! $self->table_active() ){
+        $log->info("Table $table_name not active yet. Waiting ".$self->creation_wait_time()." second");
+        if( $self->creation_wait_time() ){
+            sleep($self->creation_wait_time());
+        }
+    }
+    $log->info("Table $table_name ACTIVE. All is fine");
+    return $creation;
+}
+
+=head2 prepare_due_processes
+
+See L<Schedule::LongSteps::Storage>
+
+=cut
+
+sub prepare_due_processes{
+    my ($self, $options ) = @_;
+
+    $options ||= {};
+
+    my $now = DateTime->now();
+    my $run_at_day = $now->clone()->truncate( to => 'day' );
+
+    my @found_items = ();
+    my $query_output;
+    do{
+        # Query the by_run_at_day index
+        $query_output = $self->dynamo_db()->Query(
+            TableName => $self->table_name(),
+            IndexName => 'by_run_at_day',
+            ConsistentRead => 0, # Consistent read are not supported on secondary indices.
+            ExpressionAttributeValues => {
+                ":run_at_day" => { "S" => substr( $run_at_day->iso8601() , 0 , 10 ) },
```

Comment:
would it not be safer (and a little more readable) if you used '$dt->strftime'?

Reply:
could use ymd actually :)